iocore/aio/P_AIO.h

```diff
@@ -130,6 +105,25 @@ struct AIO_Reqs {
 };
 
 #endif // AIO_MODE == AIO_MODE_NATIVE
+
+TS_INLINE int
+AIOCallbackInternal::io_complete(int event, void *data)
+{
+  (void)event;
+  (void)data;
+  if (aio_err_callbck && !ok()) {
+    AIOCallback *err_op          = new AIOCallbackInternal();
+    err_op->aiocb.aio_fildes     = this->aiocb.aio_fildes;
+    err_op->aiocb.aio_lio_opcode = this->aiocb.aio_lio_opcode;
+    err_op->mutex                = aio_err_callbck->mutex;
+    err_op->action               = aio_err_callbck;
+    eventProcessor.schedule_imm(err_op);
+  }
+  if (!action.cancelled)
```

Comment:
You are combining the to io_complete() together. In the original native AIO one, it has the process to acquire a mutex.  Do we need it here?

Reply:
It does with `DiskHandler::mainAIOEvent`.
To call back any `handleEvent()`, It is required to get the mutex locked.

Reply:
I see. Thanks!